applications/FSIapplication/python_scripts/partitioned_embedded_fsi_base_solver.py

```diff
@@ -0,0 +1,665 @@
+        str_interface_submodelpart = self.model.GetModelPart(self._get_structure_interface_model_part_name())
+
+        # Set the INTERFACE flag to the structure skin
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.INTERFACE, True, str_interface_submodelpart.Nodes)
+
+    def _update_level_set(self):
+        # Recompute the distance field with the obtained solution
+        self._get_distance_to_skin_process().Execute()
+
+        # Correct the distance field
+        self._get_distance_modification_process().Execute()
+
+    def _solve_fluid(self):
+        # Update the current iteration level-set position
+        self._update_level_set()
+
+        # Solve fluid problem
+        self.fluid_solver.SolveSolutionStep() # This contains the FM-ALE operations
+
+        # Undo the FM-ALE virtual mesh movement
+        self.fluid_solver._get_mesh_moving_util().UndoMeshMovement()
+
+        # Restore the fluid node fixity to its original status
+        self._get_distance_modification_process().ExecuteFinalizeSolutionStep()
+
+    def _transfer_fluid_load(self):
+        if (self.level_set_type == "continuous"):
+            # Interpolate the pressure to the fluid FSI coupling interface
+            self._get_partitioned_fsi_utilities().EmbeddedPressureToPositiveFacePressureInterpolator(
+                self.GetFluidComputingModelPart(),
+                self._get_fsi_coupling_interface_fluid().GetInterfaceModelPart())
+
+            # Map PRESSURE from fluid FSI coupling interface to structure FSI coupling interface
+            # Note that in here we take advantage of the fact that the coupling interfaces coincide in the embedded case
+            KratosMultiphysics.VariableUtils().CopyModelPartNodalVar(
+                KratosMultiphysics.PRESSURE,
+                KratosMultiphysics.POSITIVE_FACE_PRESSURE,
+                self._get_fsi_coupling_interface_fluid().GetInterfaceModelPart(),
+                self._get_fsi_coupling_interface_structure().GetInterfaceModelPart(),
+                0)
+
+        elif (self.level_set_type == "discontinuous"):
+            # Generate the intersections skin to map from
+            self._get_embedded_skin_utility().GenerateSkin()
+
+            # Interpolate POSITIVE_FACE_PRESSURE and NEGATIVE_FACE_PRESSURE from the background mesh
+            self._get_embedded_skin_utility().InterpolateDiscontinuousMeshVariableToSkin(
+                KratosMultiphysics.PRESSURE, KratosMultiphysics.POSITIVE_FACE_PRESSURE, "positive")
+            self._get_embedded_skin_utility().InterpolateDiscontinuousMeshVariableToSkin(
+                KratosMultiphysics.PRESSURE, KratosMultiphysics.NEGATIVE_FACE_PRESSURE, "negative")
+
+            mapper_params = KratosMultiphysics.Parameters("""{
+                "mapper_type": "nearest_element",
+                "echo_level" : 0
+            }""")
+            mapper = KratosMapping.MapperFactory.CreateMapper(
```

Comment:
do you have to recreate it each time?
There would also be `UpdateInterface`

Reply:
The point is that the embedded skin model part is changing its entities (nodes and conditions) each time the level set is updated. Thus I think it is needed to recreate it each time. Am I wrong?

Reply:
recreating it is certainly not wrong, but might be less efficient than using `UpdateInterface`
=> for now it does the same in the end, so you can leave it as is